### PR TITLE
Test and fix return types for PostgreSQL custom range operators

### LINKED ISSUE #5476: TypeDecorator on PostgreSQL range type is passed `bool` from comparison operator.
**Describe the bug**
When defining a `TypeDecorator` based on `dialects.postgresql.ranges.INT8RANGE` (or any other range, I think), `process_result_value` is called with the result of calling the `overlaps` method (or other comparisons) from `dialects.postgresql.ranges.RangeComparison.comparison_factory`, even though that returns `bool` rather than a range type.

[Docs](https://docs.sqlalchemy.org/en/13/core/custom_types.html#redefining-and-creating-new-operators) suggest that operators that return `bool` should be marked by passing `is_comparison` to `Operators.op`, which is indeed missing (e.g. https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/dialects/postgresql/ranges.py#L58), and I _suspect_ that explains what I'm seeing.

**Expected behavior**
Calling `overlaps` or other `bool`-returning comparison methods in a query should bypass the `TypeDecorator`'s result processing.

**To Reproduce**
I am happy to provide a minimal example upon request, but would prefer to spend that time writing an actual test in a PR instead, if I can first get confirmation here that my diagnosis is correct.

**Versions.**
 - OS: Linux (Mint 20.0)
 - Python: 3.7
 - SQLAlchemy: 1.3.17 (_if_ my diagnosis is correct, the problem is present on master as well)
 - Database: PostgreSQL
 - DBAPI: psycopg2 2.8.5

### PR DESCRIPTION
**Pull request by TallJimbo**
### Description
This adds a failing test and fix for the return types of PostgreSQL-specific range operators, which return booleans in the database but were exposed to SQLAlchemy as if they returned range objects.

### Checklist
This pull request is:

- [ ] A documentation / typographical error fix
- [x] A short code fix

  - Fix is just adding `is_comparison=True` to some custom operators.
  - This should fix #5476, though the tests are at a lower-level than the problem reported there, in order to involve just compilation rather than execution; the previous discrepancy with the [return types documented by PostgreSQL](https://www.postgresql.org/docs/12/functions-range.html) (and strongly implied by the operator names) is still pretty obvious.

- [ ] A new feature implementation

**Have a nice day!**

### CODE REVIEW

File: test/dialect/postgresql/test_types.py [hunk 1]
```diff
@@ -2628,112 +2628,149 @@ def setup_class(cls):
...
+    def _test_clause(self, colclause, expected, type_):
         self.assert_compile(colclause, expected)
+        assert colclause.type is type_
 
     def test_where_equal(self):
         self._test_clause(
-            self.col == self._data_str, "data_table.range = %(range_1)s"
+            self.col == self._data_str,
+            "data_table.range = %(range_1)s",
+            sqltypes.BOOLEANTYPE,
```
Comment (zzzeek) [inline test/dialect/postgresql/test_types.py:2639]: yeah that's perfect, checks the type, great.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision 925b117e0c91cdd67d9ddbd9d65f5ca3e88af91f of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
New Gerrit review created for change 925b117e0c91cdd67d9ddbd9d65f5ca3e88af91f: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2107

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

openstack-recheck

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2107

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

Code-Review+2 Workflow+2

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2107

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2107 has been **merged**. Congratulations! :)

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

Verified+1 Code-Review+2 Workflow+2

build failures are jenkins blinking out a little bit, it needs to be seen if this is a systemic issue

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2108

**Comment by sqla-tester** [comment]
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2108 has been **merged**. Congratulations! :)

**Pull request closed without merging**